src/Compilers/Core/Portable/DiagnosticAnalyzer/DefaultAnalyzerAssemblyLoader.cs

```diff
@@ -57,12 +61,16 @@ protected override string PrepareSatelliteAssemblyToLoad(string assemblyFilePath
         /// </summary>
         /// <param name="windowsShadowPath">A shadow copy path will be created on Windows and this value 
         /// will be the base directory where shadow copy assemblies are stored. </param>
+#if NET
+        internal static IAnalyzerAssemblyLoaderInternal CreateNonLockingLoader(AssemblyLoadContext? loadContext, string windowsShadowPath, ImmutableArray<IAnalyzerAssemblyResolver>? externalResolvers = null)
+#else
         internal static IAnalyzerAssemblyLoaderInternal CreateNonLockingLoader(string windowsShadowPath, ImmutableArray<IAnalyzerAssemblyResolver>? externalResolvers = null)
```

Comment:
Think it's much easier to just have two overloads where only the one taking `ALC` is available in `#if NET`. That keep sthe calling code a lot cleaner. Basically use `#endif` where you have `#else`.

Reply:
can you clarify?  won't callers still need to know which tehy call, depending on if they're .netcore or not?

Reply:
Take a look at this as an example

https://github.com/dotnet/roslyn/blob/main/src/Compilers/Core/Portable/DiagnosticAnalyzer/ShadowCopyAnalyzerAssemblyLoader.cs#L46-L56

Reply:
i genuinely don't love that.  1. it seems duplicative.  2. It means in a .net core env, someone might call the wrong overload.  At least on the IDE side of hte fence, i want to make sure all callers must consider and pass in the ALC.  It would have to be an explicit chocie to pass in 'null' if htey want that.

Reply:
note: i'll follow whatever pattern you guys want ont he compiler side :)  so i'll make this change here.  it's just not something i intend to take into the IDE side :)

Reply:
my 2 cents: The amount of #if'ing going on here is more complicated than the code that would be duplicated by just separating out the implementations completely. This seems more readable to me:

```
#if NET
        internal static IAnalyzerAssemblyLoaderInternal CreateNonLockingLoader(string windowsShadowPath, ImmutableArray<IAnalyzerAssemblyResolver>? externalResolvers = null)
            => CreateNonLockingLoader(loadContext: null, windowsShadowPath, externalResolvers);

        internal static IAnalyzerAssemblyLoaderInternal CreateNonLockingLoader(AssemblyLoadContext? loadContext, string windowsShadowPath, ImmutableArray<IAnalyzerAssemblyResolver>? externalResolvers = null)
        {
            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
            {
                return new DefaultAnalyzerAssemblyLoader(loadContext, loadOption: AnalyzerLoadOption.LoadFromStream, externalResolvers: externalResolvers);
            }

            VerifyPathRooted(windowsShadowPath);

            return new ShadowCopyAnalyzerAssemblyLoader(loadContext, windowsShadowPath, externalResolvers);
        }
#else
        internal static IAnalyzerAssemblyLoaderInternal CreateNonLockingLoader(string windowsShadowPath, ImmutableArray<IAnalyzerAssemblyResolver>? externalResolvers = null)
        {
            VerifyPathRooted(windowsShadowPath);

            return new ShadowCopyAnalyzerAssemblyLoader(windowsShadowPath, externalResolvers);
        }
#endif

        internal static void VerifyPathRooted(string windowsShadowPath)
        {
            // The shadow copy analyzer should only be created on Windows. To create on Linux we cannot use 
            // GetTempPath as it's not per-user. Generally there is no need as LoadFromStream achieves the same
            // effect
            if (!Path.IsPathRooted(windowsShadowPath))
            {
                throw new ArgumentException("Must be a full path.", nameof(windowsShadowPath));
            }
        }
```